db/models/articles.model.js

```diff
@@ -1,26 +1,48 @@
 const db = require('../connection');
 
-function idCheck(rows) {
-    if(rows.length !== 0){
+function idCheck(id, rows) {
+    const values = [id]
+   if (rows.length === 0) {
+    return db.query(`SELECT * FROM articles WHERE article_id = $1;`, values).then(({rows}) => {
         return rows
-    }
-    else return {Error : "ID Does Not Exist"}
+    })
+    .then((rows) => {
+        if(rows.length !== 0){
+            return {'comments' : []}
+        }
+        else {
+            return {Error : "ID Does Not Exist"}
```

Comment:
Ideally we should be making use of Promise.reject as a custom error handler and grabbing the error in our catch, then passing it to our error handling middleware. Otherwise the catch is a bit redundant here and we aren't utilising the functionality of promises. I would suggest refactoring this.